subsys/bluetooth/host/adv.c

```diff
@@ -1458,6 +1458,12 @@ int bt_le_adv_stop(void)
 	return 0;
 }
 
+bool bt_le_is_advertising()
+{
+	struct bt_le_ext_adv *adv = bt_le_adv_lookup_legacy();
+	return adv && atomic_test_bit(adv->flags, BT_ADV_ENABLED);
+}
```

Comment:
Hmm, I get the point, but I think this is actually a bit misleading. 

The while the name matches the legacy advertising API, it is also generic enough that users would expect it to be returning true if an extended advertising set it advertising, which the function currently doesn't. 

Your commit message should contain the reasoning for adding this function. 

What is the use case here?

Reply:
> The while the name matches the legacy advertising API, it is also generic enough that users would expect it to be returning true if an extended advertising set it advertising, which the function currently doesn't.

Fair enough. Would the preferred approach be to support extended advertising as well, or to make it clear by documentation/function name that this function is only for legacy advertising?

> What is the use case here?

Good question. We want to keep track of our advertising state. For instance, we'd like to avoid calling bt_le_adv_update_data(...) when we're connected, since it will just return `-EAGAIN`. But perhaps the preferred usage is to call bt_le_adv_update_data(...) and simply handle the error?

Reply:
I suspect you'll need to handle the error either way, since your thread could get preempted and a connection could happen after `bt_le_is_advertising()` has returned but before you have the chance to do anything with the return value. That said, if you're not doing it already, maybe it'd be useful to register a `bt_conn_cb`, so that you know when you're connected or not?

Reply:
> Fair enough. Would the preferred approach be to support extended advertising as well, or to make it clear by documentation/function name that this function is only for legacy advertising?

Not sure. Extended advertising requires the application to create the set first, and thus the application is (or should) always aware of the advertising state of the specific set, so it's not really a use case for extended advertising IMO. 

The only time where such a function like this is really useful is when legacy advertising is used and there are multiple applications. If there is a single application, it should also be aware of the advertising state. 

> Good question. We want to keep track of our advertising state. For instance, we'd like to avoid calling bt_le_adv_update_data(...) when we're connected, since it will just return -EAGAIN. 

Not sure I understand. When you say "when we're connected" do you mean "when we are advertising connectable"? The advertising state and connection state are completely separate. 

> But perhaps the preferred usage is to call bt_le_adv_update_data(...) and simply handle the error?
`bt_le_adv_update_data` will always return `-EAGAIN` when advertising is enabled. If you only have a single application using advertising, then you should be able to track the advertising in your application fairly easily. If this is your own use case, then you can, as you thought yourself, use the return value of `-EAGAIN` to determine if you need to stop the advertising before changing the data. 

I'm not strictly opposed to adding this as an API, but I'm not sure I think your usecase is a strong enough case for it. If it was a more trivial API (it is a bit tricky with multiple advertising sets), then it would have been easier to approve :)

Reply:
>I'm not strictly opposed to adding this as an API, but I'm not sure I think your usecase is a strong enough case for it.

I'm not entirely sure either! I'm open to the fact that I might just not understand the way things are supposed to work fully.

>Not sure I understand. When you say "when we're connected" do you mean "when we are advertising connectable"? The advertising state and connection state are completely separate.

Are they really completely separate? When we establish a connection, advertising stops automatically. When we disconnect, advertising resumes automatically. And when we are connected, bt_le_adv_update_data(...) returns -EAGAIN.

>Extended advertising requires the application to create the set first, and thus the application is (or should) always aware of the advertising state of the specific set, so it's not really a use case for extended advertising IMO.

I'm not sure that I agree that it is possible to be completely aware of the advertising state. For instance, as I said, advertising is paused when a connection is established. We can keep track of that specific state change with the `connected` callback in `bt_conn_cb`, so that is fine for our intents and purposes. However, the `disconnected` callback in `bt_conn_cb` is called _before_ advertising resumes. We want to update advertising data ASAP after a disconnect, but we can only guess how long it will take for advertising to resume after a disconnect.

All in all, I'm not 100% sure that this change is the _best_ way to solve our main issue, but I do think there is an issue (and I really appreciate you taking the time on this @Thalley).

Reply:
> Are they really completely separate? When we establish a connection, advertising stops automatically.

That is true. When a connection happens on a connectable advertising set, the advertising set stops, so there is some dependency there. However, the advertising data has nothing to do with the connected state.

> When we disconnect, advertising resumes automatically.

That was actually deprecated by https://github.com/zephyrproject-rtos/zephyr/pull/73395 :) So in the future this won't be the case. The stack will no longer auto-resume advertising, which is why your application should be knowledge of the state already, but I do agree that on older versions of Zephyr this was not the case. My apologies for not considering older versions. 

I think this API makes sense for versions without https://github.com/zephyrproject-rtos/zephyr/pull/73395, but after this PR it makes less sense, as the application is (rightfully) in full control of the advertising state :)

Reply:
Thank you, now I understand perfectly! I will close this PR.